lib/collection.go

```diff
@@ -71,7 +72,35 @@ func (collectionImpl) List(scope scope, p *ListParams) ([]dsref.VersionInfo, err
 			Offset:  p.Offset,
 			Limit:   p.Limit,
 		}
-		return s.List(scope.ctx, scope.ActiveProfile().ID, lp)
+		infos, err := s.List(scope.ctx, scope.ActiveProfile().ID, lp)
+		if err != nil {
+			return nil, err
+		}
+
+		if p.EnsureFSIExists {
```

Comment:
I really dislike this param on the `ListParams`, feels very out of place and is tied to internal implementation details and should probably be something configuration specific not request param specific. I would at least rename it to `OnlyLocal` or something similar.

Reply:
I fully agree and intend to remove this entirely in a follow up PR that removes FSI before the next release